api/rql/internal/primary/meta/schema.go

```diff
@@ -0,0 +1,99 @@
+package meta
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/ekinanp/jsonschema"
+	"github.com/puppetlabs/wash/plugin"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+// Schema represents an entry metadata schema. It is a wrapper for
+// plugin.JSONSchema.
+type Schema struct {
+	loader gojsonschema.JSONLoader
+}
+
+type SchemaPredicate = func(Schema) bool
+
+func newSchema(s *plugin.JSONSchema) Schema {
+	// Given a value schema like ".key1.key2 5", the schema predicate's job is to check
+	// that m['key1']['key2'] == primitive_value. To do this correctly, we need to munge s
+	// a bit before returning our schema instance. That's what this code does.
+	var mungeProperties func(map[string]*jsonschema.Type) map[string]*jsonschema.Type
```

Comment:
Not technically needed because `mungeProperties` isn't recursive.

Reply:
Yes, but opted to keep it because code looks a little prettier.